src/main/java/tayoo/Parser.java

```diff
@@ -0,0 +1,257 @@
+package tayoo;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.temporal.ChronoField;
+import java.time.temporal.TemporalAccessor;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+import tayoo.command.*;
+import tayoo.exception.ParserException;
+import tayoo.tasks.Deadline;
+import tayoo.tasks.Event;
+import tayoo.tasks.Task;
+import tayoo.tasks.ToDo;
+
+/**
+ * The Parser class contains all the methods related to parsing. This includes parsing user commands, dates, time and
+ * text input. All methods within this class should be static.
+ */
+public class Parser {
+
+    /**
+     * Contains all the commands which can be used to exit the Tayoo chatbot.
+     */
+    private static final String[] exitCodes = {"EXIT", "BYE", "GOODBYE", "CLOSE", "QUIT"};
+
+    /**
+     * Parses the user's input. Currently utilises a series of if-else checks to correlate the user's input with any
+     * commands that have been implemented. The case of the input command is arbitrary as all commands are automatically
+     * converted to upper case. This means that if implementing any new commands, they should be in all upper case when
+     * being added to the if-else chain.
+     *
+     * @param command The user's input in a String format. The command should start with a substring representing the
+     *                type of command,
+     * @return An object that is a subclass of the Command class.
+     * @throws ParserException if there is an error when parsing a command, either because the command is not supported,
+     * or the format of the command is wrong
+     */
+    public static Command parseCommand(String command) throws ParserException {
```

Comment:
I think that you can consider naming the method like a verb so commandParser instead of parseCommand